src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/PackageSearchMetadataCacheItem.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using NuGet.Shared;
+using NuGet.Versioning;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    internal sealed class PackageSearchMetadataCacheItem
+    {
+        private IPackageSearchMetadata _packageSearchMetadata;
+        private readonly IPackageMetadataProvider _packageMetadataProvider;
+        private readonly Dictionary<NuGetVersion, PackageSearchMetadataCacheItemEntry> _cachedItemEntries;
+
+        public PackageSearchMetadataCacheItem(IPackageSearchMetadata packageSearchMetadata, IPackageMetadataProvider packageMetadataProvider)
+        {
+            _cachedItemEntries = new Dictionary<NuGetVersion, PackageSearchMetadataCacheItemEntry>
+            {
+                [packageSearchMetadata.Identity.Version] = new PackageSearchMetadataCacheItemEntry(packageSearchMetadata, packageMetadataProvider)
+            };
+
+            _packageSearchMetadata = packageSearchMetadata;
+            _packageMetadataProvider = packageMetadataProvider;
+            AllVersionsContextInfo = GetVersionInfoContextInfoAsync();
```

Comment:
Consider this instead to eliminate code duplication:
```C#
_packageMetadataProvider = packageMetadataProvider;
RefreshVersionInfoAsync(packageSearchMetadata);
```

Reply:
This will cause the code analyzer to give an error as I need to assign _packageSearchMetadata or make it nullable